If we have a serialport connection open in the background and try to open it in StepBro, an unexpected exception may occur

This is specifically the case when we have the --print_report option enabled.

```
GRTControllerDACTest.sbs --execute GRTCTestDAC --model FormalTest --exitcode SubVerdict --format AzureLog --print_report
##[group] Test Suite Test Setup
  Description: The test configuration and preconditions. Setup procedure: GRTControllerDACTest.DACPreTest.
    No data.
##[endgroup]

Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at StepBro.Core.Addons.OutputAzurePipelineFormatAddon.Outputter.WriteReport(DataReport report) in C:\SW_development\StepBro\source\StepBro.Core\Addons\OutputAzurePipelineFormatAddon.cs:line 165
   at StepBro.Cmd.Program.Main(String[] args) in C:\SW_development\StepBro\source\stepbro\Program.cs:line 496
```

If we remove the --print_report and add -v instead we see the following:

```
GRTControllerDACTest.sbs --execute GRTCTestDAC --model FormalTest --exitcode SubVerdict --format AzureLog --v
StepBro console application. Type 'stepbro --help' to show the help text.
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.CAN.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.IronPython.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.Streams.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.TestInterface.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.Process.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\StepBro.PanelCreator.dll
Assembly loaded: c:\SW_development\StepBro\source\bin\Modules\TestModule.dll
Filename: C:\SW_development\ESW-Platform\esw-platform\test\systest\GRTController\GRTControllerDACTest.sbs
               0.000 StepBro - Main logger created
##[command]               0.077    StepBro - Successfully read the station properties file.
##[section]               0.149    StepBro.Main.FileParsing - Starting file parsing
               3.172       StepBro.Main.FileParsing - Variable portGRTController - Create data
               3.173    portGRTController - Create SerialPort
               3.174       StepBro.Main.FileParsing - Variable portGRTController init: Reset and initialize, data: { Device=GRTControllerPort, BaudRate=460800, StopBits=One }
               3.175       StepBro.Main.FileParsing - Variable controller - Create data
               3.175       StepBro.Main.FileParsing - Variable controller init: Reset and initialize, data: { Stream=portGRTController, CommandResponseTimeout=00:00:02 }
               3.177       StepBro.Main.FileParsing - Ended file parsing. No errors.
##[section]               3.183    Script Execution - TestFramework.TestSuiteFormalTestExecution
               3.184       TestFramework.TestSuiteFormalTestExecution - ( 1018 TestList GRTCTestDAC )
               3.186          189 DataReport.StartGroup - Starting report group "Test Suite Test Setup". The test configuration and preconditions. Setup procedure: GRTControllerDACTest.DACPreTest.
##[section]               3.187          GRTControllerDACTest.DACPreTest - PRETEST - <no arguments>
               3.187             36 controller.Connect - Open (COM9)
               3.195             36 controller.Connect - Available port(s): 'COM3', 'COM14', 'COM13', 'COM9'
##[error]      3.196             Error - 36 - Could not open stream; Access to the path 'COM9' is denied.
               3.197       Script execution ended. Result: Error, Failure in called procedure "GRTControllerDACTest.DACPreTest".
Procedure execution ended. Result: Error, Failure in called procedure "GRTControllerDACTest.DACPreTest".
```

Showing that we got an error because of the inability to open a stream.

--- a/source/StepBro.Core/Addons/OutputAzurePipelineFormatAddon.cs
+++ b/source/StepBro.Core/Addons/OutputAzurePipelineFormatAddon.cs
@@ -180,7 +180,7 @@ namespace StepBro.Core.Addons
                             m_writer.WriteLine("##[endgroup]");
                             m_writer.WriteLine("");     // Empty line
                         }
-                        else if (group.LogEnd == null)
+                        else if (group.LogEnd == null) // If group.LogEnd is null, a fatal error has occurred during execution
                         {
                             m_writer.WriteLine("##[error] An error has occurred. Dumping log.");
                             var entry = group.LogStart;

--- a/source/StepBro.Core/Addons/OutputAzurePipelineFormatAddon.cs
+++ b/source/StepBro.Core/Addons/OutputAzurePipelineFormatAddon.cs
@@ -162,7 +162,7 @@ namespace StepBro.Core.Addons
                         m_writer.WriteLine("##[endgroup]");
                         m_writer.WriteLine("");     // Empty line
 
-                        if (group.LogEnd.Id != group.LogStart.Id)
+                        if (group.LogEnd != null && group.LogEnd.Id != group.LogStart.Id)
                         {
                             m_writer.WriteLine("##[group] " + group.Name + " - execution log");
                             indent.Push(indent.Peek() + "    ");
@@ -179,6 +179,16 @@ namespace StepBro.Core.Addons
                             indent.Pop();
                             m_writer.WriteLine("##[endgroup]");
                             m_writer.WriteLine("");     // Empty line
+                        }
+                        else if (group.LogEnd == null)
+                        {
+                            m_writer.WriteLine("##[error] An error has occurred. Dumping log.");
+                            var entry = group.LogStart;
+                            while (entry != null)
+                            {
+                                WriteLogEntry(entry, entry.Timestamp);
+                                entry = entry.Next;
+                            }
                         }
                     }
                 }

--- a/source/StepBro.Core/Addons/OutputConsoleWithColorsAddon.cs
+++ b/source/StepBro.Core/Addons/OutputConsoleWithColorsAddon.cs
@@ -83,7 +83,7 @@ namespace StepBro.Core.Addons
 
             public void WriteReport(DataReport report)
             {
-                throw new NotImplementedException();
+                throw new NotImplementedException("Test reports are not yet supported for console format.");
             }
 
             void ITextWriter.Write(string text)

